mongo/doc/bson.go

```diff
@@ -66,6 +66,10 @@ func DocumentFromStruct(
 		field := fields.Field(i)
 		value := s.Field(i)
 		key, valFace, set := valueFromStructField(field, value)
+		if key == "inline" {
+			doc = append(doc, DocumentFromStruct(valFace)...)
+			continue
+		}
```

Comment:
This is wrong. What if I have a struct like this?
```go
struct{
    Inline struct{
        FieldWithNameInline string `bson:"inline"`
    }`bson:"key_name,inline"`
}
```
IOW, this should be moved to the tag parsing logic inside `valueFromStructField` and note that the first tag ALWAYS gives the field name.

Reply:
Oops, nevermind. Should've read the bson doc first. I just struck me that a key name and the inline option doesn't make sense :facepalm: